### ISSUE TITLE
Issue in barplot

### ISSUE THREAD

**Issue by ta94** [microsoft/interpret#30]
Hi,
 In ebm.explain_global() in categorical feature plots (barplots) the x axis values calculated seems wrong to me. For categorical extra nan is coming in 'specific' 'names' list, this leads to the length mismatch between 'names' and 'scores' the last categorical value is being ignored from the plot. 
And there is no null values in my data.
For example, 
`{'density': {'names': [nan, 'abc', 'efg'],
  'scores': [66486, 118521]},
 'lower_bounds': [-2.246691779337516, 1.1945678479167394],
 'names': [nan, 'abc', 'efg'],
 'scores': [-2.136401943158369, 1.1984443228864698],
 'type': 'univariate',
 'upper_bounds': [-2.026112106979222, 1.2023207978562003]}`

**Comment by interpret-ml** [comment]
Hi @ta94,

Thanks for getting in touch. What operating system and Python version are you running on?

**Comment by gabriel6280** [comment]
I'm not sure if this is related, but I also noticed an issue related to the names. For the y-axis of  my horizontal barplots for LIME, the scores and values match, but the names do not. 

Here's an example:

```python
import interpret
import numpy as np
from interpret import show   

# loading the data
from sklearn.datasets import load_wine

wine_data = load_wine()
X_wine = wine_data['data']
y_wine = wine_data['target']
wine_data_feature_names = wine_data['feature_names']


# make wine data set binary
# from:to
wine_class_conversion_dict = {0:0, 1:1, 2:0}
y_wine = np.array([wine_class_conversion_dict[class_num] for class_num in y_wine])
np.unique(y_wine)


# train/test split
from sklearn.model_selection import train_test_split

proportion_of_data_for_testing = 0.25
X_wine_train, X_wine_test, y_wine_train, y_wine_test = train_test_split(X_wine, y_wine, test_size=proportion_of_data_for_testing)


# fit a blackbox
from sklearn.ensemble import RandomForestClassifier
rfc_wine = RandomForestClassifier(n_estimators=32)
rfc_wine.fit(X_wine_train, y_wine_train)


# use LIME on that RFC
from interpret.blackbox import LimeTabular

#Blackbox explainers need a predict function, and optionally a dataset
lime = LimeTabular(predict_fn=rfc_wine.predict_proba,
                   data=X_wine_train, random_state=1, feature_names=wine_data_feature_names)

# Pick the instances to explain, optionally pass in labels if you have them
lime_local = lime.explain_local(X_wine_test[:5], y_wine_test[:5], name='LIME')


show(lime_local)  # LOOK at idx=0
## values and scores correspond, but the names on the y-axis do not

##### LOOK at idx=0 for above dashboard and below dictionary

## what the values should actually be
print('row at idx 0 in test set')
print({feature_name:val for feature_name, val in zip(lime_local.data(0)['names'], lime_local.data(0)['values'])})


## demonstrating that show() is, somewhere, using 
# lime_local.feature_names rather than lime_local.data(0)['names']
from copy import deepcopy

lime_local_example = deepcopy(lime_local)
lime_local_example.feature_names = ['asdf' for _ in range(0, len(lime_local_example.feature_names))]
show(lime_local_example)
```

Perhaps I'm misunderstanding something?

**Comment by interpret-ml** [comment]
Hi @gabriel6280 - that doesn't sound good, it's likely on our end.

There was a recent bug that was surfaced around v0.1.12 relating to incorrect graphs. Can you try updating to the latest version v0.1.13 and see if it's the same result?

**Comment by gabriel6280** [comment]
> Hi @gabriel6280 - that doesn't sound good, it's likely on our end.
> 
> There was a recent bug that was surfaced around v0.1.12 relating to incorrect graphs. Can you try updating to the latest version v0.1.13 and see if it's the same result?

Just did that, and now, everything seems to match-up correctly. Thanks a lot!
